x-pack/elastic-agent/pkg/agent/application/pipeline/dispatcher/dispatcher.go

```diff
@@ -74,7 +76,22 @@ func (ad *ActionDispatcher) key(a fleetapi.Action) string {
 }
 
 // Dispatch dispatches an action using pre-registered set of handlers.
-func (ad *ActionDispatcher) Dispatch(acker store.FleetAcker, actions ...fleetapi.Action) error {
+// ctx is used here ONLY to carry the span, for cancelation use the cancel
+// function of the ActionDispatcher.ctx.
+func (ad *ActionDispatcher) Dispatch(ctx context.Context, acker store.FleetAcker, actions ...fleetapi.Action) (err error) {
+	span, ctx := apm.StartSpan(ctx, "dispatch", "app.internal")
```

Comment:
[Blocker]
@stuartnelson3 the span isn't being ended anymore or capturing the error. Was it intentional?

Reply:
That was my mistake from cargo-culting your code suggestion above, i'll add it back